cpp/test/sg/fil_test.cu

```diff
@@ -1040,6 +1040,21 @@ std::vector<FilTestParams> predict_dense_inputs = {
                   algo        = BATCH_TREE_REORG,
                   leaf_algo   = CATEGORICAL_LEAF,
                   num_classes = 3),
+  // use shared memory opt-in carveout if available, or infer out of L1 cache
+  FIL_TEST_PARAMS(num_rows = 103, num_cols = ((48 + 1) * 1024) / sizeof(float), algo = NAIVE),
+  FIL_TEST_PARAMS(num_rows    = 103,
+                  num_cols    = ((48 + 1) * 1024) / sizeof(float),
```

Comment:
It would be better to base these numbers on `MAX_SHM_STD`, e.g. `(MAX_SHM_STD + 1024) / sizeof(float)`.